src/Tasks/system.design/stronglytypedresourcebuilder.cs

```diff
@@ -106,19 +107,17 @@ internal static CodeCompileUnit Create(IDictionary resourceList, String baseName
             var resourceTypes = new Dictionary<String, ResourceData>(StringComparer.InvariantCultureIgnoreCase);
             foreach (DictionaryEntry de in resourceList)
             {
-                var node = de.Value as ResXDataNode;
```

Comment:
Is there no longer any case that the `resourceList` will contain a `ResXDataNode`?

Reply:
No, I don't think so. The only place I see things added to the `resourcesHashTable` that gets passed here is:

https://github.com/microsoft/msbuild/blob/8684c1203903b243adf89374013081803286200f/src/Tasks/GenerateResource.cs#L3842-L3851

which explicitly requires an `IResource`.

Before merging this I think I'll try to change to a generic dictionary to validate that. Not sure how impactful that would be yet.

Reply:
It was as easy as it could possibly have been.

Reply:
Now that this is an `IResource` does it make sense to lift `TypeName` up to the interface and avoid a downcast?

Reply:
I'm doing so in a follow-up PR that fixes this problem for Core scenarios too. We can do it all at once if you prefer.

Reply:
I see, up to you.